pkg/operator/controller/controller.go

```diff
@@ -111,6 +116,19 @@ func (r *ReconcileCDI) Reconcile(request reconcile.Request) (reconcile.Result, e
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling CDI")
 
+	//TODO:
+	/*
+		Here for upgrade purpose:
+		* if cr vresion observed and target are not the same upgrade is taking place:
```

Comment:
maybe a stupid question: where/how is target version set?

Reply:
TargetVersion is set initially on crInit  this is in case Get of CRD CR did not find the object
``` func (r *ReconcileCDI) crInit(cr *cdiv1alpha1.CDI) error {
	cr.Finalizers = append(cr.Finalizers, finalizerName)
	cr.Status.OperatorVersion = r.namespacedArgs.DockerTag
	cr.Status.TargetVersion = r.namespacedArgs.DockerTag
	return r.crUpdate(cdiv1alpha1.CDIPhaseDeploying, cr)
}
``` 
In case CT was found - in Reconcile method we check whether it is upgrade by comparing cr.ObeservedVersion to DockerTag environmental variable. If they are not the same - cr is updated with cr.TargetVersion = DockerTag and from that point isUpgrading method of reconcile object will return true.
When current operator is deploied and ready, cr.Observed is set to cr.Target

Reply:
This is a common pattern and looks good.